core/domain/topic_services.py

```diff
@@ -1342,6 +1342,29 @@ def check_can_edit_topic(
     return False
 
 
+def check_can_edit_question(
+    user: user_domain.UserActionsInfo,
+    topic_rights: Optional[topic_domain.TopicRights]
+) -> bool:
+    """Checks whether the user can edit the questions.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        topic_rights: TopicRights or None. Rights object for the given topic.
+
+    Returns:
+        bool. Whether the given user can edit the question.
+    """
+    if topic_rights is None:
+        return False
+    if role_services.ACTION_EDIT_ANY_QUESTION in user.actions:
+        return True
+    if role_services.ACTION_EDIT_QUESTION_IN_MANAGED_TOPIC in user.actions:
+        return check_can_edit_topic(user, topic_rights)
+    return False
```

Comment:
The `return False`/`True` statements can be a bit hard to read. If we have some code like this:

```python
if condition_A:
  return True
return False
```
we generally would like to refactor it to:
```python
return condition_A
```
This does the same thing as the first snippet, but it's much shorter and it encourages developers to think more in terms of the condition of the `if` statement as opposed to the result of that condition being met.

Reply:
Hi, I don't think I understood correctly what you wanted me to alter. Is it something like this?

![image](https://github.com/oppia/oppia/assets/52565678/c1294365-ae63-4a44-915f-51a4be4066cd)

Reply:
Alter the entire code snippet. In this case, it helps to start refactoring from the bottom and move upwards. So let's focus on this snippet:
```python
if role_services.ACTION_EDIT_QUESTION_IN_MANAGED_TOPIC in user.actions:
    return check_can_edit_topic(user, topic_rights)
return False
```
We want to return False `if role_services.ACTION_EDIT_QUESTION_IN_MANAGED_TOPIC not in user.actions`. So we should have:
```python
return (
    role_services.ACTION_EDIT_QUESTION_IN_MANAGED_TOPIC in user.actions and
    check_can_edit_topic(user, topic_rights)
)
```
as `and` will evaluate to False without checking the remaining conditions if one of the conditions evaluate to False.

Now take a look at the if statement right above it:
```python
if role_services.ACTION_EDIT_ANY_QUESTION in user.actions:
    return True
```
Since we're returning `True` already, we don't have to rewrite the resulting boolean. We want to return this early `if role_services.ACTION_EDIT_ANY_QUESTION in user.actions`, so we should connect this to what we've refactored so far using `or`. This boolean operator returns True without checking the remaining conditions if one of the conditions evaluates to True. So, we now have:
```python
return role_services.ACTION_EDIT_ANY_QUESTION in user.actions or (
    role_services.ACTION_EDIT_QUESTION_IN_MANAGED_TOPIC in user.actions and
    check_can_edit_topic(user, topic_rights)
)
```

Now for the top `if` statement, we consider how this fits into what we've refactored so far. We want to return False early `if topic_rights is None`. So we should achieve this with `and`. So our final statement would be:
```python
return topic_rights is not None and (
    role_services.ACTION_EDIT_ANY_QUESTION in user.actions or (
        role_services.ACTION_EDIT_QUESTION_IN_MANAGED_TOPIC in user.actions and
        check_can_edit_topic(user, topic_rights)
    )
)
```

Reply:
Done!